### ISSUE TITLE
ntransfcufftplan vs ntransf

### ISSUE THREAD

**Issue by garrettwrong** [flatironinstitute/cufinufft#25]
I was struggling to get parity between the `many` methods in ASPIRE, and of course, I was using the wrong variable (oops :sweat_smile: ), so only yielding the first transforms worth or results. That is totally operator error, but I would like to understand this a little better.

Can anyone tell me if ntransfcufftplan is required for me as an end user?  I did some light reading and I see lines like below, and also the related PR #20 ... It appears this is to control size of loop of batches? I'm guessing there is a limit to the simultaneous transforms? Do we have a good default value you can recommend?

`for(int i=0; i*d_plan->ntransfcufftplan < d_plan->ntransf; `

**Comment by MelodyShih** [comment]
Hi Garrett, I just saw this. I will look into it and also the other issue about gpu_method==3 gives wrong result tonight. Could you let me know the typical size of the transformation (i.e. N1, N2) ASPIRE usually has or the one you’re running into trouble? Thanks.

**Comment by MelodyShih** [comment]
A quick answer would be for the end user, one doesn’t have to worry about ntransfcufftplan ( changed to maxbatchsize) and ntransf is number of data to apply the nonuniform FFT. A potential issue to keep in mind is that larger maxbatchsize means more GPU memory required.

**Comment by garrettwrong** [comment]
For 2d2 many, testing data is 1024 transforms of 64x64.  I think we are targeting 256x256 for higher res. For sake of discussion say also 1024 transforms.  IIRC on my personal gpu I had to run small batches, maybe 128 to fit.

I guess the main question is whether this parameter is more for performance, or just making things fit?  

The second question would be, should the sanity check default be 1, maxbatchsize=ntransf, or something else. I don't mind putting basic logic on our side, or tuning, but a good baseline idea will help me a lot.  I just got the code to reproduce, so while I am cleaning it is a good time to add any size logic.

The main trouble was I was only setting the batch size instead of ntransf, but that was my trouble, not your trouble :).
